test.py

```diff
@@ -1,13 +1,54 @@
-from flake8_print import check_code_for_print_statements, PRINT_ERROR_CODE, PRINT_ERROR_MESSAGE
+from textwrap import dedent
+
+from pep8 import BaseReport, Checker, StyleGuide
+from flake8_print import print_invocations
 from nose.tools import assert_equal, assert_true
 
 
-# Python 3 detects the col as the column with the execution brackets rather than the function name. So we will attempt both when necessary.
+class CaptureReport(BaseReport):
+    """Collect the results of the checks."""
+
+    def __init__(self, options):
+        self._results = []
+        super(CaptureReport, self).__init__(options)
+
+    def error(self, line_number, offset, text, check):
+        """Store each error."""
+        code = super(CaptureReport, self).error(line_number, offset,
+                                                 text, check)
+        if code:
+            self._results.append(
+                (line_number, offset, code, text[5:], check.__doc__))
+        return code
+
+
+class PrintTestStyleGuide(StyleGuide):
+
+    logical_checks = [
+        ('print_invocations', print_invocations, ['logical_line', 'tokens', 'noqa']),
+    ]
+    physical_checks = []
+    ast_checks = []
+    max_line_length = None
+    hang_closing = False
+    verbose = False
+    benchmark_keys = {'files': 0, 'physical lines': 0, 'logical lines': 0}
+
+
+_print_test_style = PrintTestStyleGuide()
+
+
+def check_code_for_print_statements(code):
+    """Process code using pep8 Checker and return all errors."""
+    report = CaptureReport(options=_print_test_style)
+    lines = [line + '\n' for line in code.split('\n')]
+    checker = Checker(lines=lines, options=_print_test_style, report=report)
+    checker.check_all()
+    return report._results
 
 
 class Flake8PrintTestCases(object):
-    def generate_error_statement(self, line, col):
```

Comment:
`generate_error_statement` was unused; if there is an intention to use it in the future, I can re-add it.